libs/nx-esbuild/src/executors/build/executor.ts

```diff
@@ -10,9 +11,16 @@ export default async function runExecutor(
     if (!context.projectName) {
         throw new Error('No projectName')
     }
-    const libRoot = context.workspace.projects[context.projectName].root
+    const appRoot = context.workspace.projects[context.projectName].root
+    const tree = new FsTree(context.cwd, context.isVerbose)
 
-    const args = getEsbuildArgs(options, libRoot)
+    const packageJson = readJson(tree, `${appRoot}/package.json`)
+
+    const args = getEsbuildArgs(
+        options,
+        appRoot,
+        Object.keys(packageJson?.dependencies || {}),
```

Comment:
rather than a third arg can you not add Object.keys(packageJson?.dependencies || {}) to options.externals

Reply:
I could, I added an additional because `options.externals` is user specified, the package json dependencies is automatically added to externals (not a default, always added)